src/components/Reservation.js

```diff
@@ -1,9 +1,238 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import DatePicker from 'react-datepicker';
+import 'react-datepicker/dist/react-datepicker.css';
+import { Navigate, useNavigate } from 'react-router-dom';
+import moment from 'moment';
+import {
+  Form, Button, Row, Col,
+} from 'react-bootstrap';
+import { useSelector, useDispatch } from 'react-redux';
+import styles from '../styles/reservation.module.css';
+import postReservationThunk from '../helpers/postReservation';
+import { isTokenExpired } from '../helpers/authUser';
+import { logOutUser } from '../redux/Auth/auth';
+import { clearStatus } from '../redux/Reservations/reservation';
+
+export default function Reservation(props) {
+  const username = useSelector((state) => state.user.username);
+  const token = useSelector((state) => state.user.token);
+  const regStatus = useSelector((state) => state.reservations);
+  const destinations = useSelector((state) => state.destinations.destinations);
+
+  const [startDate, setStartDate] = useState(null);
+  const [endDate, setEndDate] = useState(null);
+  const [submitted, setSubmitted] = useState(false);
+  let { destination } = props;
+  const destinationPassed = destination !== 'default';
+  destination = destination === 'default' ? destinations[0].name : destination;
+  const [chosenDestination, setChosenDestination] = useState(destination);
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  console.log(chosenDestination);
+
+  useEffect(() => {
+    const res = isTokenExpired();
+    if (res) {
+      localStorage.removeItem('token');
+      dispatch(logOutUser());
+      navigate('/login');
+    }
+    dispatch(clearStatus());
+  }, []);
+
```

Comment:
Hey @themonster2015 ,
I just figured out that this hook makes my side to redirect to the login page, and considering that I am logged in it redirects me to the home page each time I try to use the  `/reserve` path. That's mean that your i`sTokenExpired` function returns true, in spite it should return false. Can you fix it, please?